src/RedirectMiddleware.php

```diff
@@ -98,7 +99,18 @@ public function checkRedirect(
             );
         }
 
-        return $this($nextRequest, $options);
+        /** @var PromiseInterface|ResponseInterface $response */
+        $options['_previous_uris'][] = $request->getUri();
+        $response = $this($nextRequest, $options);
+
+        // Add headers to be able to track history of redirects.
+        if (!empty($options['allow_redirects']['track_redirects'])) {
+            $response = $response->wait(TRUE);
+            if (!$response->hasHeader('X-Guzzle-Redirect')) {
```

Comment:
Instead of this, just use the `withAddedHeader` method of a response and remove the if statement.